hist/hist/src/TF1.cxx

```diff
@@ -532,7 +532,7 @@ TF1::TF1(const char *name, const char *formula, Double_t xmin, Double_t xmax, EA
    // Create rep formula (no need to add to gROOT list since we will add the TF1 object)
 
    // First check if we are making a convolution
-   if (TString(formula, 5) == "CONV(" && formula[strlen(formula) - 1] == ')') {
+   if (TString(formula, std::min(strlen(formula), 5)) == "CONV(" && formula[strlen(formula) - 1] == ')') {
```

Comment:
We should call `strlen` only once and cache the result.  Also, we need to add an explicit check that `formula`.  Have a look further down the method, there is the `NSUM(` check which should be treated in the same way and there are a couple of more invocations of `strlen(formula)` which can be replaced by the cached result.

Reply:
Okay. I will have a look at this. I may need some time as my exams have started.

Reply:
@jblomer  so if I understand correctly, instead of repeatedly calling strlen(formula), we save the result? Could we use a variable to do this, because I'm not familiar with NSUM and can't find anything online. Maybe you can send me some resources on that and I can familiarize myself.

Reply:
To be honest, it's a bit stupid to construct a `TString` to check if the string starts with `"CONV("`.
How about
```c++
if (strncmp(formula, "CONV(", 5) == 0 && formula[strlen(formula) - 1] == ')' ) {
```
?

Note also that `strncmp` stops at `\0` characters, so you don't need to check the length beforehand.

Reply:
But Jakob is right that a
```c++
const auto formulaLength = strlen(formula);
```
Can save computation time.

Reply:
```suggestion
   if (!strncmp(formula, "CONV(", 5) && formula[strlen(formula) - 1] == ')') {
```

or

```suggestion
   TString sFormula = formula;
   if (sFormula(0, 5) == "CONV(" && sFormula.EndsWith(")")) {
```